libraries/AP_Logger/AP_Logger_File.cpp

```diff
@@ -715,6 +709,36 @@ void AP_Logger_File::stop_logging(void)
     }
 }
 
+/*
+  does start_new_log in the logger thread
+ */
+void AP_Logger_File::PrepForArming_start_logging()
+{
+    if (logging_started()) {
+        return;
+    }
+
+    uint32_t start_ms = AP_HAL::millis();
+    const uint32_t open_limit_ms = 1000;
+
+    /*
+      log open happens in the io_timer thread. We allow for a maximum
+      of 1s to complete the open
+     */
+    start_new_log_pending = true;
+    EXPECT_DELAY_MS(1000);
+    while (AP_HAL::millis() - start_ms < open_limit_ms) {
+        if (logging_started()) {
+            break;
+        }
+#if !APM_BUILD_TYPE(APM_BUILD_Replay) && !defined(HAL_BUILD_AP_PERIPH)
+        // keep the EKF ticking over
+        AP::ahrs().update();
```

Comment:
I guess this doesn't update the sensors though so some data the EKF could need won't be there.  For example the visual odometry libraries aren't updated so the EKF would still likely become unhappy if the user is doing non-gps navigation.  Not a blocker of course, just pointing out that it probably doesn't cover all situations.

Reply:
yes, but it is the high rate IMU data that really matters for it diverging badly. It is the big time steps
usually the file open will happen in 0.1s or so